make/autoconf/flags-ldflags.m4

```diff
@@ -86,7 +86,7 @@ AC_DEFUN([FLAGS_SETUP_LDFLAGS_HELPER],
     LDFLAGS_CXX_PARTIAL_LINKING="$MACHINE_FLAG -r"
 
     if test "x$OPENJDK_TARGET_OS" = xaix; then
-      BASIC_LDFLAGS="-Wl,-b64 -Wl,-brtl -Wl,-bnorwexec -Wl,-bnolibpath -Wl,-bnoexpall \
+      BASIC_LDFLAGS="-Wl,-b64 -Wl,-brtl -Wl,-bnolibpath -Wl,-bnoexpall \
```

Comment:
We don't use clang on AIX, so this should make any difference (yet?).

Reply:
yes, i think we switched to xlC16's xlclang already.  for java23, OpenXL(LLVM based) is expected.

Reply:
No, we're still using the xlc toolchain; see the most recent acceptance build: https://openj9-jenkins.osuosl.org/job/Build_JDKnext_ppc64_aix_OpenJDK/672.
```
[2024-05-29T04:26:54.412Z] configure: Using default toolchain xlc (IBM XL C/C++)
```

Reply:
see later:
`BUILD_ENV_VARS_LIST = [PATH+XLC=/opt/IBM/xlC/16.1.0/bin:/opt/IBM/xlc/16.1.0/bin, CC=xlclang, CXX=xlclang++]
`
That is clang too. plus, the head guard is `==xaix`, and it is an linker option (linker is the same between xlC and OpenXL). It has the same effect.

Reply:
The presence of "clang" in the names of the tools isn't relevant, the toolchain name, "xlc", is the reason the changed code is unused (see line 82):
```
  elif test "x$TOOLCHAIN_TYPE" = xclang; then
```